assets/js/components/Dashboard/NewPaMessage/AssociateAlertPage.tsx

```diff
@@ -0,0 +1,347 @@
+            />
+            <div className="subtext">Locations will be editable</div>
+          </div>
+          <div className="checkbox">
+            <Form.Check
+              id="import-message-checkbox"
+              label="Import message"
+              checked={importMessage}
+              onChange={() => {
+                setImportMessage(!importMessage);
+              }}
+            />
+            <div className="subtext">Message will be editable</div>
+          </div>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            onClick={() => {
+              setAssociatedAlert({} as Alert);
+              setShowAlertModal(false);
+            }}
+            className="cancel-button"
+            variant="link"
+          >
+            Cancel
+          </Button>
+          <Button
+            onClick={() => {
+              if (importMessage) {
+                setVisualText(associatedAlert.header);
+              }
+              navigateTo(Page.NEW);
+            }}
+            className="apply-button"
+          >
+            Apply
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    </div>
+  );
+};
+
+interface AssociateAlertsTableProps {
+  alerts: Alert[];
+  messageStateFilter: string;
+  serviceTypeFilter: string;
+  setAssociatedAlert: Dispatch<SetStateAction<Alert>>;
+  setShowAlertModal: Dispatch<SetStateAction<boolean>>;
+}
+
+const AssociateAlertsTable: ComponentType<AssociateAlertsTableProps> = ({
+  alerts,
+  messageStateFilter,
+  serviceTypeFilter,
+  setAssociatedAlert,
+  setShowAlertModal,
+}: AssociateAlertsTableProps) => {
+  const filterByActiveState = (alert: Alert) => {
+    const alertStart = new Date(alert.active_period[0].start);
+    const currentTime = new Date();
+    return messageStateFilter === "active"
+      ? alertStart <= currentTime
+      : alertStart > currentTime;
+  };
+
+  const filterByServiceType = (alert: Alert) => {
+    return serviceTypeFilter === "All"
+      ? true
+      : alert.affected_list.some((affected) =>
+          affected.includes(serviceTypeFilter.toLowerCase()),
+        );
+  };
+
+  const filteredAlerts = alerts.filter((alert) => {
+    return filterByActiveState(alert) && filterByServiceType(alert);
+  });
+
+  return (
+    <>
+      <table className="associate-alert-table">
+        <thead>
+          <tr>
+            <th className="associate-alert-table__message">Alert message</th>
+            <th className="associate-alert-table__id">ID</th>
+            <th className="associate-alert-table__start-end">Start-End</th>
+            <th className="associate-alert-table__last-modified">
+              Last modified
+            </th>
+            <th className="associate-alert-table__select"></th>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredAlerts.length == 0 ? (
+            <tr>
+              <div className="associate-alert-table__empty">
```

Comment:
Noticed a console error here. I think direct children of `tr` tags need to be either `th` or `td` elements. Should be able to just wrap the `div` in a `td`.